spec/controllers/users_controller_spec.rb

```diff
@@ -243,6 +241,41 @@
       delete :purge_avatar, params: { id: user.id }
       expect(user.reload.avatar).not_to be_attached
     end
+
+    it 'doesnt allow a user to change their own role' do
+      updated_params = {
+        role_id: create(:role, name: 'New Role').id
+      }
+      patch :update, params: { id: user.id, user: updated_params }
+
+      user.reload
+
+      expect(user.role_id).not_to eq(updated_params[:role_id])
+    end
+
+    it 'allows an admin to edit themself' do
```

Comment:
Instead I'd say, "allows a user with ManageUsers permission to edit their own role"